fesvr/htif.cc

```diff
@@ -129,7 +129,7 @@ std::map<std::string, uint64_t> htif_t::load_payload(const std::string& payload,
   } preload_aware_memif(this);
 
   try {
-    return load_elf(path.c_str(), &preload_aware_memif, entry);
+    return load_elf(path.c_str(), &preload_aware_memif, entry, is32bit);
```

Comment:
I don't like passing variables by non-const ref because it's not clear at the call site here that `is32bit` will be modified by the function.

Why don't you pass an `unsigned required_xlen` and do the checking inside `load_elf()`?

Maybe inside `load_elf()` isn't the right place to print error messages, so throw a new exception and catch it where appropriate.

Reply:
Agreed

Reply:
(But my previous comment about non-Spike use cases still stands: after taking Scott’s suggestion into account, only perform the check / throw the exception if xlen has actually been set)

Reply:
Done: so I only check if xlen was actually set, and no more passing refs: I only pass required_xlen to load_elf and that is also an optional arg so if you don't pass it it wouldn't check (e.g. for elf2hex), and throw an exception if there is an issue.